src/test/ui/conditional-compilation/cfg_accessible-not_sure.rs

```diff
@@ -0,0 +1,85 @@
+// revisions: edition2015 edition2021
+// [edition2015]compile-flags: --edition=2015
+// [edition2021]compile-flags: --edition=2021
+
+#![feature(extern_types)]
+#![feature(cfg_accessible)]
+
+// Struct::unresolved - error
+
+struct Struct {
+    a: u8,
+    pub b: u8
+}
+
+#[cfg_accessible(Struct::a)] //~ ERROR not sure
+const A: bool = true;
+#[cfg_accessible(Struct::b)] //~ ERROR not sure
+const B: bool = true;
+#[cfg_accessible(Struct::c)] //~ ERROR not sure
+const C: bool = true;
+
+// Union::unresolved - error
+
+struct Union {
+    a: u8,
+    pub b: u8
+}
+
+#[cfg_accessible(Union::a)] //~ ERROR not sure
+const A: bool = true;
+#[cfg_accessible(Union::b)] //~ ERROR not sure
+const B: bool = true;
+#[cfg_accessible(Union::c)] //~ ERROR not sure
+const C: bool = true;
+
+// Enum::unresolved - error
+
+enum Enum {
+    A(u8),
+}
+
+#[cfg_accessible(Enum::B)] //~ ERROR not sure
+const B: bool = true;
+
+// Trait::unresolved - false or error, depending on edition (error if you can write Trait::foo
+// instead of <dyn Trait>::foo for methods like impl dyn Trait { fn foo() {} })
+
+trait Trait {}
+impl dyn Trait { fn foo() {} }
+
+// FIXME: Should be a error for edition > 2015
+#[cfg_accessible(Trait::foo)] //~ ERROR not sure
+const A: bool = true;
+
+// TypeAlias::unresolved - error
+
+type TypeAlias = Struct;
+
+#[cfg_accessible(TypeAlias::A)] //~ ERROR not sure
+const A: bool = true;
+
+// ForeignType::unresolved - error
+
+extern {
+    type ForeignType;
+}
+
+#[cfg_accessible(ForeignType::A)] //~ ERROR not sure
+const A: bool = true;
+
+// AssocType::unresolved - error
+
+trait AssocType {
+    type AssocType;
+}
+
+#[cfg_accessible(AssocType::AssocType::A)] //~ ERROR not sure
+const A: bool = true;
+
+// PrimitiveType::unresolved - error
+
+#[cfg_accessible(u8::A)] //~ ERROR not sure
+const A: bool = true;
+
+fn main() {}
```

Comment:
Missing test cases:
- Variant::unresolved - error (for compatibility with variant types)
- Trait::unresolved (actually unresolved, not an existing method from `dyn Trait`)

Reply:
Also, the same set of test cases, but with existing names, like
```rust
Struct::existing
Enum::ExistingVariant
...
```
, they should all produce "not sure" errors except for `Enum::ExistingVariant` and `Trait::existing_assoc_item` which produce true.